packages/mars/src/options/config.ts

```diff
@@ -30,28 +30,31 @@ export async function getConfig(options: Options): Promise<ExecuteOptions> {
     }
   }
 
-  const { wallet, networkName } = await getWallet(merged)
-  const gasPrice = merged.gasPrice ?? (await wallet.provider.getGasPrice())
+  const { signer, networkName } = await getSigner(merged)
+  const gasPrice = merged.gasPrice ?? (await signer.getGasPrice())
 
   return {
     gasPrice,
     noConfirm: !!merged.noConfirm,
-    wallet,
-    network: networkName,
+    signer,
+    networkName: networkName,
     dryRun: !!merged.dryRun,
     logFile: merged.logFile ?? '',
     deploymentsFile: merged.outputFile,
     verification,
   }
 }
 
-async function getWallet(options: Options) {
-  const { network, infuraApiKey, alchemyApiKey, dryRun, privateKey } = options
+async function getSigner(options: Options) {
+  const { network, infuraApiKey, alchemyApiKey, dryRun, fromAddress, privateKey } = options
   if (network === undefined) {
     throw new Error('No network specified. This should never happen.')
   }
   let rpcUrl
-  if (network.startsWith('http')) {
+  let provider
+  if (typeof network === 'object') {
```

Comment:
I'd extract it as a type guard to make it more clear what's going on

Reply:
Btw, `typeof null === 'object'` is true so we should double check for the case when there's no provider. (yes, I know it's gonna be undefined bot still let's be safe here)

Reply:
> I'd extract it as a type guard to make it more clear what's going on

You rather meant type predicate, didn't you? https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

Reply:
I opened #50 to address the issue for network ambiguity in broader scope.